plugins/router/template/router.go

```diff
@@ -64,6 +68,13 @@ type templateRouter struct {
 	statsPassword string
 	// if the router can expose statistics it should expose them with this port
 	statsPort int
+	// rateLimitedCommitFunction is a rate limited commit (persist state + refresh the backend)
+	// function that coalesces and controls how often the router is reloaded.
+	rateLimitedCommitFunction *ratelimiter.RateLimitedFunction
+	// rateLimitedCommitStopChannel is the stop/terminate channel.
+	rateLimitedCommitStopChannel chan struct{}
+
+	sync.Mutex
```

Comment:
Prefer we don't embed this, call it "lock", and add a comment here on what it covers